fastestimator/pipeline/pipeline.py

```diff
@@ -330,28 +331,25 @@ def _get_batch_size(self, epoch):
     def show_results(self, inputs=None, mode="train", num_steps=1, current_epoch=0):
         data = []
         self._prepare(inputs=inputs)
-        dataset = self.dataset_schedule[mode].get_current_value(current_epoch)
-        for example in dataset.take(num_steps):
-            data.append(example)
+        ds_iter = self.dataset_schedule[mode].get_current_value(current_epoch)
+        for _ in range(num_steps):
+            data.append(next(ds_iter))
         self._reset()
```

Comment:
this also resets the iter() object right?

Reply:
it resets everything in pipeline at the end